lib/middlewares.ts

```diff
@@ -54,11 +61,23 @@ export const jwtVerifyMiddleware = async (request: NextApiRequest) => {
   if (tokenClaims === null) {
     throw new MiddlewareError("Missing token.", 400);
   }
-  if ((tokenClaims as JwtPayload).iss !== domain) {
+  if ((tokenClaims as DashboardTokenPayload).iss !== saleorDomain) {
+    throw new MiddlewareError("Invalid token.", 400);
+  }
+
+  const client = createClient(
+    `https://${saleorDomain}/graphql/`,
+    async () => Promise.resolve({ token: getAuthToken() }),
```

Comment:
Hmm, shouldn't `getAuthToken` be an async function?
Wouldn't then 
`async () => ({ token: await getAuthToken() })`
work?

Reply:
It's not async, as we just read environment variable of content of a file (synchronously): https://github.com/saleor/saleor-app-template/blob/main/lib/environment.ts#L7